src/world.rs

```diff
@@ -236,6 +333,7 @@ impl World {
     }
 }
 
+/// An entity id, with a generation
```

Comment:
I would probably refrain from making the generation part of the documentation as it is merely an implementation detail to track stale entity handles (and could be solved in other ways). Just an "opaque entity handle" or "opaque entity identifier" seems preferable IMHO.